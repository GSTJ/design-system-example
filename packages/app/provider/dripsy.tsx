import { darkTheme, defaultThemeProps, lightTheme } from 'app/tokens'
import { DripsyFinalTheme, DripsyProvider, makeTheme } from 'dripsy'

// https://www.dripsy.xyz/usage/theming/create
export const theme = makeTheme(defaultThemeProps)

export const groupedDarkTheme = {
  ...theme,
  ...darkTheme,
}

export const groupedLightTheme = {
  ...theme,
  ...lightTheme,
}

type DripsyProps = {
  children: React.ReactNode
  theme: DripsyFinalTheme
}

export const Dripsy: React.FC<DripsyProps> = ({ children, theme }) => {
  return (
    <DripsyProvider
      theme={theme}
      // this disables SSR, since react-native-web doesn't have support for it (yet)
      ssr
    >
      {children}
    </DripsyProvider>
  )
}
