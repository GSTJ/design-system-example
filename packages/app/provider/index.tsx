import { StatusBar, useColorScheme } from 'react-native'

import { QueryClient, QueryClientProvider } from '@tanstack/react-query'

import { Dripsy, groupedDarkTheme, groupedLightTheme } from './dripsy'

export const queryClient = new QueryClient()

export const Provider = ({ children }: { children: React.ReactNode }) => {
  const colorMode = useColorScheme()
  const isDarkMode = colorMode === 'dark'

  return (
    <QueryClientProvider client={queryClient}>
      <StatusBar barStyle={isDarkMode ? 'dark-content' : 'light-content'} />
      <Dripsy theme={isDarkMode ? groupedDarkTheme : groupedLightTheme}>
        {children}
      </Dripsy>
    </QueryClientProvider>
  )
}
