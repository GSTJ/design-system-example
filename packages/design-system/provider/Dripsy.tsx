import { DripsyFinalTheme, DripsyProvider, makeTheme } from 'dripsy'
import { darkTheme, defaultThemeProps, lightTheme } from 'tokens'

// https://www.dripsy.xyz/usage/theming/create
export const theme = makeTheme(defaultThemeProps)

export const groupedDarkTheme = {
  ...theme,
  ...darkTheme,
}

export const groupedLightTheme = {
  ...theme,
  ...lightTheme,
}

type DripsyProps = {
  children: React.ReactNode
  theme: DripsyFinalTheme
}

export const Dripsy: React.FC<DripsyProps> = ({ children, theme }) => {
  return (
    <DripsyProvider
      theme={theme}
      // this disables SSR, since react-native-web doesn't have support for it (yet)
      ssr
    >
      {children}
    </DripsyProvider>
  )
}
