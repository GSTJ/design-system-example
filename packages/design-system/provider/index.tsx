import { StatusBar, useColorScheme } from 'react-native'

import { Dripsy, groupedDarkTheme, groupedLightTheme } from './dripsy'

export const Provider = ({ children }: { children: React.ReactNode }) => {
  const colorMode = useColorScheme()
  const isDarkMode = colorMode === 'dark'

  return (
    <Dripsy theme={isDarkMode ? groupedDarkTheme : groupedLightTheme}>
      <StatusBar barStyle={isDarkMode ? 'dark-content' : 'light-content'} />
      {children}
    </Dripsy>
  )
}
