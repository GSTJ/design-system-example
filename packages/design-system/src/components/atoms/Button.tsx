import React from 'react'

import { ActivityIndicator, Pressable, Text } from 'dripsy'

export interface ButtonProps extends React.ComponentProps<typeof Pressable> {
  children: React.ReactNode
  isLoading?: boolean
}

export const Button: React.FC<ButtonProps> = ({
  children,
  sx,
  isLoading,
  disabled,
  ...props
}) => (
  <Pressable
    style={({ hovered, pressed }) => ({
      opacity: hovered || pressed ? 0.7 : 1,
    })}
    sx={{
      transition: 'opacity 120ms ease-out',
      backgroundColor: '$primary',
      padding: 3,
      borderRadius: 2,
      alignItems: 'center',
      justifyContent: 'center',
      ...sx,
    }}
    disabled={isLoading || disabled}
    {...props}
  >
    <Text sx={{ fontWeight: 'bold', color: '$buttonText' }}>
      {isLoading ? (
        <ActivityIndicator size={16} color="$buttonText" />
      ) : (
        children
      )}
    </Text>
  </Pressable>
)
