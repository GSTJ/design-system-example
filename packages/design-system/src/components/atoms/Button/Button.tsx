import React from 'react'

import { ActivityIndicator, Pressable, SxProp, Text } from 'dripsy'

export interface ButtonProps extends React.ComponentProps<typeof Pressable> {
  children: React.ReactNode
  isLoading?: boolean
}

export const Button: React.FC<ButtonProps> = ({
  children,
  sx,
  isLoading,
  disabled,
  ...props
}) => (
  <Pressable
    style={({ hovered, pressed }) => ({
      opacity: hovered || pressed ? 0.7 : 1,
    })}
    sx={
      {
        backgroundColor: '$buttonBackground',
        transition: 'opacity 120ms ease-out',
        padding: 15,
        borderRadius: 2,
        alignItems: 'center',
        justifyContent: 'center',
        ...sx,
      } as SxProp
    }
    disabled={isLoading || disabled}
    {...props}
  >
    <Text
      sx={{
        fontWeight: 'bold',
        color: '$buttonText',
        fontSize: '',
      }}
    >
      {isLoading ? (
        <ActivityIndicator size={16} color="$buttonText" />
      ) : (
        children
      )}
    </Text>
  </Pressable>
)
