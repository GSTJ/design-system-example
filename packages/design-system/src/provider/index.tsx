import React from 'react'
import { StatusBar, useColorScheme } from 'react-native'

import { Dripsy, groupedDarkTheme, groupedLightTheme } from './providers/Dripsy'

export const Provider = ({
  children,
  colorMode,
}: {
  children: React.ReactNode
  colorMode?: 'dark' | 'light'
}) => {
  const colorScheme = useColorScheme()
  const activeTheme = colorMode || colorScheme

  const isDarkMode = activeTheme === 'dark'

  return (
    <Dripsy theme={isDarkMode ? groupedDarkTheme : groupedLightTheme}>
      <StatusBar barStyle={isDarkMode ? 'dark-content' : 'light-content'} />
      {children}
    </Dripsy>
  )
}
