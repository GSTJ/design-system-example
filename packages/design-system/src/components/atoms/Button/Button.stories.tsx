import React from 'react'

import type { Meta, StoryFn } from '@storybook/react'

import { Button } from './Button'

export default {
  title: 'Atoms/Button',
  component: Button,
  args: {
    children: 'Button',
  },
} as Meta<typeof Button>

export const Default: StoryFn<typeof Button> = (args) => <Button {...args} />
