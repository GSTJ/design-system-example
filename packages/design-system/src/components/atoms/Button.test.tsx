import React from 'react'
import { fireEvent, render } from '@testing-library/react-native'

import { Provider } from '../../provider'

import { Button, ButtonProps } from './Button'

type ButtonVariants = ButtonProps['variant'][]

describe('Button', () => {
  it.each(['layout.ghost', 'layout.primary'] as ButtonVariants)(
    'renders %s variant',
    (variant) => {
      const component = render(<Button variant={variant}>Press me</Button>, {
        wrapper: Provider,
      })
      expect(component).toMatchSnapshot()
    }
  )

  it('fires onPress when pressed', () => {
    const onPress = jest.fn()
    const TEST_ID = 'button-id'

    const { getByTestId } = render(
      <Button testID={TEST_ID} onPress={onPress}>
        Press me
      </Button>,
      { wrapper: Provider }
    )

    fireEvent.press(getByTestId(TEST_ID))

    expect(onPress).toHaveBeenCalled()
  })
})
