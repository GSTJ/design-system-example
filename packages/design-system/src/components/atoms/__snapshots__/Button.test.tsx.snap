// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders layout.ghost variant 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  showCursor={true}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderColor": "#FBC710",
        "borderWidth": 2,
      },
      {
        "opacity": 1,
      },
      {
        "alignItems": "center",
        "borderRadius": 2,
        "justifyContent": "center",
        "padding": 15,
        "transition": "opacity 120ms ease-out",
      },
    ]
  }
>
  <Text
    style={
      [
        {},
        undefined,
        {
          "color": "#0A0A0A",
          "fontWeight": "bold",
        },
      ]
    }
  >
    Press me
  </Text>
</View>
`;

exports[`Button renders layout.primary variant 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  showCursor={true}
  style={
    [
      {
        "backgroundColor": "#FBC710",
      },
      {
        "opacity": 1,
      },
      {
        "alignItems": "center",
        "borderRadius": 2,
        "justifyContent": "center",
        "padding": 15,
        "transition": "opacity 120ms ease-out",
      },
    ]
  }
>
  <Text
    style={
      [
        {},
        undefined,
        {
          "color": "#0A0A0A",
          "fontWeight": "bold",
        },
      ]
    }
  >
    Press me
  </Text>
</View>
`;
